src/Core/CustomMethods.php

```diff
@@ -282,10 +282,15 @@ protected function removeMethodsFrom($property, $index = null)
                 if (!isset(self::$extra_methods[$class][$method])) {
                     continue;
                 }
-                    
+
                 $methodInfo = self::$extra_methods[$class][$method];
 
-                if ($methodInfo['property'] === $property && $methodInfo['index'] === $index) {
+                if (
+                    // always check for property
+                    (isset($methodInfo['property']) && $methodInfo['property'] === $property) &&
+                    // check for index only if provided
+                    (!$index || ($index && isset($methodInfo['index']) && $methodInfo['index'] === $index))
```

Comment:
Are the falsy truthy checks necessary here? I notice they weren’t there originally

Reply:
I think so.
`$property` is a required param of the function, so we can check it directly.
`$index` is optional, so we only check it when it's provided and when there's a corresponding item in the `$methodInfo` array.

The second way how this can be done is
`((!isset($methodInfo['index']) || (isset($methodInfo['index']) && $methodInfo['index'] === $index))`.

Open to suggestions on how to further simplify this.

Reply:
Right, but the logic change is basically preventing the array value from being what the default value of the optional variable is. Just curious if the lack of that is causing a problem or not right now

Reply:
The _isset_ checks aren't sufficient, the existing unit tests fail in that case.

Reply:
I see, you're talking about `ViewableDataTest::testSetFailover` which reports that a method exists when it shouldn't?

Reply:
Yep, correct.